src/app/qgsmaptoolsplitfeatures.cpp

```diff
@@ -92,44 +92,84 @@ void QgsMapToolSplitFeatures::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
     deleteTempRubberBand();
 
     //bring up dialog if a split was not possible (polygon) or only done once (line)
-    int topologicalEditing = QgsProject::instance()->topologicalEditing();
+    bool topologicalEditing = QgsProject::instance()->topologicalEditing();
+    QgsPointSequence topologyTestPoints;
     vlayer->beginEditCommand( tr( "Features split" ) );
-    QgsGeometry::OperationResult returnCode = vlayer->splitFeatures( captureCurve(), true, topologicalEditing );
-    vlayer->endEditCommand();
-    if ( returnCode == QgsGeometry::OperationResult::NothingHappened )
-    {
-      QgisApp::instance()->messageBar()->pushMessage(
-        tr( "No features were split" ),
-        tr( "If there are selected features, the split tool only applies to those. If you would like to split all features under the split line, clear the selection." ),
-        Qgis::Warning,
-        QgisApp::instance()->messageTimeout() );
-    }
-    else if ( returnCode == QgsGeometry::OperationResult::GeometryEngineError )
-    {
-      QgisApp::instance()->messageBar()->pushMessage(
-        tr( "No feature split done" ),
-        tr( "Cut edges detected. Make sure the line splits features into multiple parts." ),
-        Qgis::Warning,
-        QgisApp::instance()->messageTimeout() );
-    }
-    else if ( returnCode == QgsGeometry::OperationResult::InvalidBaseGeometry )
-    {
-      QgisApp::instance()->messageBar()->pushMessage(
-        tr( "No feature split done" ),
-        tr( "The geometry is invalid. Please repair before trying to split it." ),
-        Qgis::Warning,
-        QgisApp::instance()->messageTimeout() );
-    }
-    else if ( returnCode != QgsGeometry::OperationResult::Success )
+    QgsGeometry::OperationResult returnCode = vlayer->splitFeatures( captureCurve(), topologyTestPoints, false, topologicalEditing );
+
+    switch ( returnCode )
```

Comment:
`returnCode` is `OperationResult::Success` even if no features were split. This change is a step closer to fixing #38287 but further changes are eventually needed.